server/portal/apps/onboarding/steps/tas_project_system_creation.py

```diff
@@ -0,0 +1,139 @@
+from portal.apps.onboarding.steps.abstract import AbstractStep
+from portal.apps.onboarding.state import SetupState
+from portal.apps.webhooks.callback import WebhookCallback
+from portal.apps.tas_project_systems.utils import get_tas_project_system_variables
+from portal.apps.system_creation.utils import (
+    call_reactor,
+)
+from portal.apps.onboarding.execute import (
+    execute_setup_steps,
+)
+from portal.libs.agave.models.systems.storage import StorageSystem
+from requests.exceptions import RequestException
+import json
+import logging
+
+
+class TasProjectSystemCreationStep(AbstractStep):
+    logger = logging.getLogger(__name__)
+
+    def __init__(self, user):
+        """
+        Call super class constructor
+        """
+        super(TasProjectSystemCreationStep, self).__init__(user)
+
+    def display_name(self):
+        return "TAS Project Systems"
+
+    def description(self):
+        return "Setting up access to data files on the storage systems based on your TAS Projects. No action required."
+
+    def prepare(self):
+        self.state = SetupState.PENDING
+        self.log("Awaiting storage system creation")
+
+    def process(self):
+        tas_project_systems = get_tas_project_system_variables(self.user, force=True)
+
+        self.logger.debug("KeyService variables substituted: {}".format(tas_project_systems))
+
+        # Create only storage systems that are not currently accessible
+        systemList = []
+        for systemId in tas_project_systems.keys():
+            try:
+                success, _ = StorageSystem(self.user.agave_oauth.client, id=systemId, load=False).test()
+            except RequestException as err:
+                self.logger.error("Test of system '{}' failed unexpectedly! Listing of system returned: {}".format(self.id, str(err)))
+                # we will assume that system needs to be created
+                success = False
+
+            if success:
+                self.logger.info(
+                    "{username} has valid configuration for {systemId}, skipping creation".format(
+                        username=self.user.username, systemId=systemId
+                    )
+                )
+            else:
+                systemList.append(systemId)
+
+        # If all required systems already exist, mark this step complete
+        if len(systemList) == 0:
+            self.complete("Found existing storage systems")
+            return
+
+        # Store requested systemIds
+        data = {
+            'requested': systemList,
+            'failed': [],
+            'successful': []
+        }
+
+        for systemId, variables in tas_project_systems.items():
```

Comment:
Here all systems are going to be recreated even if just a few need to be created.  

(I noticed the same issue in `key_service_creation.py` when working on #513 so just wanted to point it out here. Is this a replacement for `key_service_creation.py`..like will that be removed from code base or exist along this step?)